api/api_test.go

```diff
@@ -608,12 +609,17 @@ func TestAPI(t *testing.T) {
 		}},
 		{"Manual scan with invalid credential", func(t *testing.T) {
 			pubkey := pk1.PublicKey()
-			badAuthClient := api.NewClient("http://"+listenAddr+"/api", "")
 			_, err := badAuthClient.ScanHost(pubkey)
 			if err == nil {
 				t.Fatal("got no error when trying to use manual scan with bad auth")
 			}
 		}},
+		{"Syncer connect with invalid credential", func(t *testing.T) {
+			err := badAuthClient.SyncerConnect("127.0.0.0.1:65535")
+			if err == nil {
```

Comment:
When making sure that something fails for a given reason, make sure to check what the reason was. Either by checking against a specific exposed error, status code (should be 401) or at least a string comparison if the error is not exposed.